src/foam/mlang/mlang.js

```diff
@@ -3225,6 +3225,49 @@ foam.CLASS({
 
   documentation: 'A Binary Predicate which applies arg2.f() to arg1.f().',
 
+  properties: [
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'arg1'
+    },
+    {
+      class: 'foam.mlang.ExprProperty',
+      name: 'arg2'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'f',
+      code: function(o) {
+        return this.arg2.f(this.arg1.f(o));
+      },
+      javaCode: `
+        Object e1 = getArg1().f(obj);
+        if ( e1 == null ) return e1;
+        return getArg2().f(e1);
+      `
+    },
+
+    function comparePropertyValues(o1, o2) {
+      /**
+         Compare property values using arg2's property value comparator.
+         Used by GroupBy
+      **/
+      return this.arg2.comparePropertyValues(o1, o2);
+    }
+  ]
+});
+
+
+foam.CLASS({
```

Comment:
Can you move this into its own source file, please - we are starting to break up these large js files.

Reply:
fixed